### ISSUE TITLE
Difference between .so and .pyd file?

### ISSUE THREAD

**Issue by mebner98** [Huelse/SEAL-Python#126]
I am using your seal-python wrapper successfully for my application in combination with https://github.com/s0l0ist/node-seal for my angular frontend. If i start everything locally on my windows machine both versions work together fine. But now I am trying to deploy my architecture into docker and therefor build your seal-wrapper for linux instead of the .pyd file. The problem is, that SEAL gives me the error RuntimeError: loaded SEALHeader is invalid now when i load a cipher (which got created in the frontend) into SEAL in python. Do you know why that could happen? Are there version differences between the linux and windows build?

cipher = Ciphertext()
    # create temp file to load cipher text into seal
    with tempfile.NamedTemporaryFile(delete=False) as temp_file:
        temp_file.write(cipher_bytes)
        temp_file_path = temp_file.name

    cipher.load(context, temp_file_path)
    os.remove(temp_file_path)
    return cipher
    
    error gets thrown at cipher.load

As i said locally everything works fine but dockerization creates that problem

**Comment by Huelse** [comment]
The .so is for linux, .pyd is for windows.

**Comment by mebner98** [comment]
yes i know! So when i am running the python flask application on my windows pc locally i use the .pyd file and it works. But as soon i put the python flask application in a linux docker container (and therefor use the .so file as seen in the posted dockerfile) i get the error with the invalid seal header. If i run everything in docker except for the flask application locally on my pc (--> .pyd file) than it works too.

It seems like the compability problems are a result of the .so file usage in the docker container but i have to use it

**Comment by mebner98** [comment]
I solved it - the problem was a result of the difference in the compression modes. 
In your build specification you build the windows file without -DSEAL_USE_ZSTD=OFF but for linux it is stated.
I removed it for the linux build and now it works
